service/src/event/service.rs

```diff
@@ -307,6 +402,71 @@ pub enum InvalidItem {
     },
 }
 
+#[async_trait]
+impl Store for CeramicEventService {
+    async fn local_sourced_data_events(&self) -> anyhow::Result<Vec<AnchorRequest>> {
+        // We must have a node DID to use here. Return an error if we don't.
+        let node_did = self
+            .node_did
+            .clone()
+            .ok_or_else(|| Error::new_app(anyhow!("node DID required to get anchor requests")))?;
+
+        // Fetch event CIDs from the events table using the previous high water mark
+        Ok(
+            CeramicOneEvent::unanchored_events_from_high_water_mark(
```

Comment:
I'm a bit worried that reading 1M is going to be slow.. but not nearly as worried as I am about writing them back.